Add a loading indicator

dmfalke opened the pull request:
This addresses https://github.com/VEuPathDB/web-eda/issues/171. Presumably, this will be desirable everywhere this component is used.

jtlong3rd reviewed (commented):
This approach makes sense! I can't think of a case where we wouldn't want to render a `Loading` while the approval status is being checked.

One gotcha - currently, the JSX returned in the case where `approvalStatus` is `loading`, `approved`, or `not-approved` all have a similar structure. This is intentional:

```typescript
<div /* some approvalStatus-dependent props */>
  {children}
</div>
```

The reason for this common structure is to prevent, during reconciliation, the `children` from being unmounted-and-remounted when the `approvalStatus` changes. This unmounting-and-remounting is potentially undesirable, as it (1) resets the `children`'s component state, (2) triggers effect cleanup(s), etc.

Since this PR changes the structure of the JSX returned when `approvalStatus` is `loading`, I would also update the JSX for when `approvalStatus` is `approved` or `not-approved` to something like this:

```typescript
<>
  <></>
  <div /* some approvalStatus-dependent props */>
    {children}
  </div>
</>
```

That should prevent unintentional unmounting-and-remounting of `children` when the `approvalStatus` changes.

([Toy project](https://codesandbox.io/s/bold-glade-pvlud?file=/src/index.tsx) to demonstrate.)

dmfalke commented:
This looks promising! I will update the branch with your suggested changes.

dmfalke commented:
I added a commit that ensures the same vdom structure is used in all cases that include `children`. I also changed how the props for the wrapper div are assigned to make it easier to keep the structure the same. I tested this with the associated EDA codebase, reverting the [commit that addressed this issue](https://github.com/VEuPathDB/web-eda/pull/403/commits/e8109b7d1a828d15bb7d7c3e254d57f746fc37ca), and it works nicely!
